lib/id3c/cli/redcap.py

```diff
@@ -254,15 +254,30 @@ class Record(dict):
     Must be constructed with a REDCap :py:class:``Project``, which is stored as
     the ``project`` attribute and used to set the ``id`` attribute storing the
     record's primary id.
+
+    Note that event_name and repeat_instance might not be populated because their
+    corresponding fields (redcap_event_name and redcap_repeat_instance)
+    won't be returned by the API if the request includes the 'fields' parameter
+    but not the record_id field.
     """
     project: Project
     id: str
+    event_name: Optional[str]
+    repeat_instance: Optional[int]
 
     def __init__(self, project: Project, data: Any = {}) -> None:
         super().__init__(data)
         self.project = project
         self.id = self[self.project.record_id_field]
```

Comment:
Ah, actually now it's apparent that this will break if the data returned from the API request doesn't include the record's primary id....

Would it be weird to ensure that the record's primary id is _always_ included in the `fields` parameter if the `fields` parameter is provided?

Reply:
I can imagine situations where that would be undesirable (e.g. when the primary id field contains PII and so is intentionally not requested), but I think for our purposes it would be reasonable to always include it. Alternatively, we could instead adjust this code in `Record` to DTRT.

Reply:
Would the right thing here mean that Record could return None for id?
It seems like raising an exception would be better since consumers would most likely assume that record.id is always set.

Reply:
Oh that's true, all of our downstream processes depend on the record's primary id to exist.

Reply:
Yes. I'd probably turn the `id` attribute into a dynamic getter so the exception only happens on access. Something like:

```python
@property
def id(self) -> str:
    """
    Returns the record's primary id.

    Raises a :py:class:`RuntimeError` if the primary id field is not available,
    usually because it was not requested from the API.
    """
    try:
        return self[self.project.record_id_field]
    except KeyError as e:
        raise RuntimeError(f"Primary record id field «{self.project.record_id_field}» not available on fetched record") from e
```

Reply:
Good idea. Don't raise an exception unless the consumer is trying to access id. I've added that in in the latest commit.

Reply:
@tsibley are you OK with what's in the final commit for this getter?